### ISSUE TITLE
Build errors under Ubuntu 18.04 with GCC 9

### ISSUE THREAD

**Issue by slhck** [tdv/redis-cpp#18]
When I build under Ubuntu 18.04, I get:

```
FAILED: src/source.p/source.cpp.o 
c++ -Isrc/source.p -Isrc -I../src -I../include/loguru -Iinclude -I../include -Isrc/test_utils -I../src/test_utils -I../include/test -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -g -pthread -MD -MQ src/source.p/source.cpp.o -MF src/source.p/source.cpp.o.d -o src/source.p/source.cpp.o -c ../src/source.cpp
In file included from ../include/redis-cpp/stream.h:33,
                 from ../src/source.cpp:37:
../include/redis-cpp/detail/stream.hpp:109:18: error: 'io_context' in namespace 'boost::asio' does not name a type
  109 |     boost::asio::io_context io_context_;
      |                  ^~~~~~~~~~
../include/redis-cpp/detail/stream.hpp:110:42: error: 'io_context_' was not declared in this scope
  110 |     boost::asio::ip::tcp::socket socket_{io_context_};
      |                                          ^~~~~~~~~~~
../include/redis-cpp/detail/stream.hpp:110:53: error: could not convert '{<expression error>}' from '<brace-enclosed initializer list>' to 'boost::asio::ip::tcp::socket' {aka 'boost::asio::basic_stream_socket<boost::asio::ip::tcp>'}
  110 |     boost::asio::ip::tcp::socket socket_{io_context_};
      |                                                     ^
      |                                                     |
      |                                                     <brace-enclosed initializer list>
../include/redis-cpp/detail/stream.hpp: In lambda function:
../include/redis-cpp/detail/stream.hpp:82:57: error: 'io_context_' was not declared in this scope
   82 |                 boost::asio::ip::tcp::resolver resolver{io_context_};
      |                                                         ^~~~~~~~~~~
../include/redis-cpp/detail/stream.hpp:82:68: error: no matching function for call to 'boost::asio::ip::basic_resolver<boost::asio::ip::tcp>::basic_resolver(<brace-enclosed initializer list>)'
   82 |                 boost::asio::ip::tcp::resolver resolver{io_context_};
      |                                                                    ^
In file included from /usr/include/boost/asio.hpp:63,
                 from ../include/redis-cpp/detail/stream.hpp:17,
                 from ../include/redis-cpp/stream.h:33,
                 from ../src/source.cpp:37:
/usr/include/boost/asio/ip/basic_resolver.hpp:67:12: note: catestdate: 'boost::asio::ip::basic_resolver<InternetProtocol, ResolverService>::basic_resolver(boost::asio::io_service&) [with InternetProtocol = boost::asio::ip::tcp; ResolverService = boost::asio::ip::resolver_service<boost::asio::ip::tcp>]'
   67 |   explicit basic_resolver(boost::asio::io_service& io_service)
      |            ^~~~~~~~~~~~~~
/usr/include/boost/asio/ip/basic_resolver.hpp:67:12: note:   conversion of argument 1 would be ill-formed:
In file included from ../include/redis-cpp/stream.h:33,
                 from ../src/source.cpp:37:
../include/redis-cpp/detail/stream.hpp:83:83: error: no matching function for call to 'boost::asio::ip::basic_resolver<boost::asio::ip::tcp>::resolve(std::remove_reference<const std::basic_string_view<char>&>::type, std::remove_reference<const std::basic_string_view<char>&>::type)'
   83 |                 auto endpoints = resolver.resolve(std::move(host), std::move(port));
      |                                                                                   ^
```

This is using the following packages installed:

```bash
apt-get update -qq && \
  apt-get install --assume-yes --no-install-recommends \
  ca-certificates \
  build-essential \
  wget \
  ninja-build \
  pkg-config \
  cmake \
  ninja-build \
  libboost-all-dev \
  libcurl4-openssl-dev \
  libpcap-dev \
  libavahi-client-dev \
  python3-pip \
  software-properties-common

add-apt-repository ppa:ubuntu-toolchain-r/test && \
  apt-get update -qq && \
  apt install --assume-yes gcc-9 g++-9 && \
  update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 60 --slave /usr/bin/g++ g++ /usr/bin/g++-9
```

So I have `g++ (Ubuntu 9.4.0-1ubuntu1~18.04) 9.4.0` and libboost-all-dev version `1.65.1.0ubuntu1`.

**Comment by slhck** [comment]
Seems that when you use the `mhier/libboost-latest` PPA and install `libboost1.74` it works.

Perhaps you want to state the minimum requirements for dependencies in the README?

**Comment by tdv** [comment]
Hi.

Thanks for the request.

I'd tried to build the library on my system with boost 1.71 (available out of the box within ubuntu 20.04) and everything worked. I've mentioned version 1.71 on the README page. I'm not sure about the minimum version, it's likely 1.67. You could try if you like and report the result.

Good luck.
Regards

**Comment by slhck** [comment]
Yeah I think it's simply the fact that 18.04 does not bundle the required libraries. But then again, it's outdated and it only comes with GCC 7 out of the box. I'll close this issue then. Thanks for your help!
